### PR TITLE
[rpc] add raw field into BlockView to support get raw block

### PR DESCRIPTION
**Pull request by coldnight**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: #3479 

## What is the new behavior?

- Add `raw` field into `GetBlockOption` to support specify it to get raw block
- Add optional `raw` field into `BlockView` and fill it when above field is set to true.

### CODE REVIEW

File: rpc/api/src/types.rs [hunk 1]
```diff
@@ -749,11 +749,17 @@ pub struct BlockView {
...
     pub uncles: Vec<BlockHeaderView>,
+    pub raw_body: Option<String>,
 }
 
 impl BlockView {
-    pub fn try_from_block(block: Block, thin: bool) -> Result<Self, anyhow::Error> {
+    pub fn try_from_block(block: Block, thin: bool, raw: bool) -> Result<Self, anyhow::Error> {
         let (header, body) = block.into_inner();
+        let raw_body = if raw {
+            Some(hex::encode(bcs_ext::to_bytes(&body)?))
```
Comment (jolestar) [inline rpc/api/src/types.rs:759]: 这里只包含 body 么？感觉 raw block 更重要的是 header，通过 header 才能计算出 block hash

Comment (coldnight) [reply rpc/api/src/types.rs:759]: 我搞混了 body hash 和 block hash，我再看下。

File: rpc/api/src/types.rs [hunk 2]
```diff
@@ -749,11 +749,20 @@ pub struct BlockView {
...
+    /// Encoded in hex format.
+    pub raw: Option<String>,
 }
 
 impl BlockView {
-    pub fn try_from_block(block: Block, thin: bool) -> Result<Self, anyhow::Error> {
+    pub fn try_from_block(block: Block, thin: bool, raw: bool) -> Result<Self, anyhow::Error> {
         let (header, body) = block.into_inner();
+        let raw_block = if raw {
+            Some(hex::encode(bcs_ext::to_bytes(&header)?))
```
Comment (jolestar) [inline rpc/api/src/types.rs:762]: raw 字段只包含了 `haader`，会不会造成误解？或者干脆把 `header` 和 `body` 分两个 raw 字段？ @sanlee42 有什么建议？

Comment (coldnight) [reply rpc/api/src/types.rs:762]: 那就提供一个单独的 `RawBlockView` 会更加清晰。

**Review by jolestar — APPROVED** [review]
LGTM

**Pull request merged**